File: app/backend/src/tv_watchlist/models/column_spec.py
"""Description of one column in a watch-order sheet."""

from __future__ import annotations

from typing import Literal

from pydantic import BaseModel, Field

ColumnKind = Literal["text", "choice"]
ColumnRole = Literal["order", "title", "watch", "other"]


class ColumnSpec(BaseModel):
    """A single sheet column, as the UI needs to render and edit it."""

    key: str
    label: str
    index: int = Field(ge=1)
    kind: ColumnKind
    role: ColumnRole
    choices: list[str] = Field(default_factory=list)
    width: float | None = None

File: app/frontend/src/types/ColumnSpec.ts
export type ColumnKind = "text" | "choice";
export type ColumnRole = "order" | "title" | "watch" | "other";

export interface ColumnSpec {
  key: string;
  label: string;
  index: number;
  kind: ColumnKind;
  role: ColumnRole;
  choices: string[];
  width: number | null;
}
